Source/GameBaseFramework/Private/Characters/Components/GBFAbilityInputBufferComponent.cpp

```diff
@@ -0,0 +1,212 @@
+#include "Characters/Components/GBFAbilityInputBufferComponent.h"
+
+#include "Characters/Components/GBFHeroComponent.h"
+#include "Characters/Components/GBFPawnExtensionComponent.h"
+#include "GAS/Components/GBFAbilitySystemComponent.h"
+#include "Input/GBFInputComponent.h"
+
+void UGBFAbilityInputBufferComponent::StartMonitoring( FGameplayTagContainer input_tags_to_check, ETriggerPriority trigger_priority )
+{
+    if ( input_tags_to_check.IsEmpty() )
+    {
+        return;
+    }
+
+    Reset();
+    TriggerPriority = trigger_priority;
+    InputTagsToCheck = input_tags_to_check;
+    BindActions();
+}
+
+void UGBFAbilityInputBufferComponent::StopMonitoring()
+{
+    RemoveBinds();
+    TryToTriggerAbility();
+    Reset();
+}
+
+void UGBFAbilityInputBufferComponent::Reset()
+{
+    TriggeredTags.Reset();
+    InputTagsToCheck.Reset();
+    BindHandles.Reset();
+}
+
+void UGBFAbilityInputBufferComponent::BindActions()
+{
+    if ( InputTagsToCheck.IsEmpty() )
+    {
+        return;
+    }
+
+    auto * pawn = GetPawn< APawn >();
+    if ( pawn == nullptr )
+    {
+        return;
+    }
+
+    const auto * hero_component = UGBFHeroComponent::FindHeroComponent( pawn );
+    if ( hero_component == nullptr )
+    {
+        return;
+    }
+
+    for ( auto & input_config : hero_component->GetBoundActionsByInputconfig() )
+    {
+        auto * input_component = pawn->FindComponentByClass< UGBFInputComponent >();
```

Comment:
no need for a find, you should just be able to access the InputComponent variable

Reply:
Done